programs/ziti-edge-tunnel/ziti-edge-tunnel.c

```diff
@@ -1818,8 +1823,13 @@ static void run(int argc, char *argv[]) {
             mask |= (ip[i] & 0xFFU);
         }
 
-        tun_ip = htonl(mask | 0x1);
-        dns_ip = htonl(mask | 0x2);
+        tun_ip = htonl(mask);
+        dns_ip = htonl(mask + 1);
+        uint32_t dns_nw_addr = mask & (0xFFFFFFFFUL << (32 - bits)) & 0xFFFFFFFFUL;
+
+        ip_addr_t dns_ip4_addr = IPADDR4_INIT(htonl(dns_nw_addr));
+        dns_range = strdup(ipaddr_ntoa(&dns_ip4_addr));
```

Comment:
`dns_range` needs to be bigger, because you're about to append the prefix to it.